contracts/stellar-interchain-token-service/src/tests/migrate.rs

```diff
@@ -0,0 +1,589 @@
+            token_id: token_id.clone(),
+            epoch: current_epoch,
+        };
+
+        legacy_storage::set_flow_in(&env, flow_key.clone(), &flow_in_amount);
+        legacy_storage::set_flow_out(&env, flow_key, &flow_out_amount);
+
+        storage::set_token_id_config(&env, token_id.clone(), &token_config);
+    });
+
+    assert_auth!(owner, its_client.upgrade(&its_wasm_hash));
+
+    let migration_data = CustomMigrationData {
+        new_token_manager_wasm_hash: token_manager_wasm_hash_v110.clone(),
+        new_interchain_token_wasm_hash: interchain_token_wasm_hash_v110.clone(),
+    };
+
+    env.mock_all_auths_allowing_non_root_auth();
+
+    env.as_contract(&its_client.address, || {
+        assert_ok!(InterchainTokenService::__migrate(&env, migration_data));
+    });
+
+    env.as_contract(&its_client.address, || {
+        assert_ok!(InterchainTokenService::migrate_token(
+            &env,
+            token_id.clone(),
+            upgrader_client.address,
+            String::from_str(&env, VERSION_V110),
+        ));
+    });
+
+    assert_eq!(
+        env.as_contract(&its_client.address, || {
+            storage::token_manager_wasm_hash(&env)
+        }),
+        token_manager_wasm_hash_v110,
+        "token manager WASM hash should be updated"
+    );
+    assert_eq!(
+        env.as_contract(&its_client.address, || {
+            storage::interchain_token_wasm_hash(&env)
+        }),
+        interchain_token_wasm_hash_v110,
+        "interchain token WASM hash should be updated"
+    );
+    assert_eq!(
+        env.as_contract(&its_client.address, || {
+            storage::flow_in(&env, token_id.clone(), current_epoch)
+        }),
+        flow_in_amount
+    );
+    assert_eq!(
+        env.as_contract(&its_client.address, || {
+            storage::flow_out(&env, token_id, current_epoch)
+        }),
+        flow_out_amount
+    );
+}
+
+#[test]
+fn migrate_token_fails_with_invalid_token_id() {
+    let (env, its_client, _, _, _) = setup_env();
+    let upgrader_client = setup_upgrader(&env);
+    let owner = its_client.owner();
+    let non_existent_token_id = BytesN::random(&env);
+
+    let its_wasm_hash = env.deployer().upload_contract_wasm(ITS_WASM);
+    let token_manager_wasm_hash_v110 = env.deployer().upload_contract_wasm(TOKEN_MANAGER_WASM_V110);
+    let interchain_token_wasm_hash_v110 = env
+        .deployer()
+        .upload_contract_wasm(INTERCHAIN_TOKEN_WASM_V110);
+
+    assert_auth!(owner, its_client.upgrade(&its_wasm_hash));
+
+    let migration_data = CustomMigrationData {
+        new_token_manager_wasm_hash: token_manager_wasm_hash_v110,
+        new_interchain_token_wasm_hash: interchain_token_wasm_hash_v110,
+    };
+
+    env.as_contract(&its_client.address, || {
+        assert_ok!(InterchainTokenService::__migrate(&env, migration_data));
+    });
+
+    assert_contract_err!(
+        its_client
+            .mock_all_auths_allowing_non_root_auth()
```

Comment:
if we see this used anywhere, we're likely introducing frontrunning vulnerability suggesting authorization isn't being handled properly

Reply:
`migrate_token` only allows the `owner` to call it, so why is non root auth needed?